src/cswinrt/code_writers.h

```diff
@@ -5499,20 +5581,65 @@ private % AsInternal(InterfaceTag<%> _) => _default;
             bind([&](writer& w)
             {
                 bool has_base_type = !std::holds_alternative<object_type>(get_type_semantics(type.Extends()));
+                if (is_fast_abi_class(type))
+                {
+                    int hierarchy_index = get_class_hierarchy_index(type);
+                    if (hierarchy_index > 0 || !type.Flags().Sealed())
+                    {
+                        auto default_interface = get_default_interface(type);
+                        auto default_iface_method_count = 0;
+                        for_typedef(w, get_type_semantics(default_interface), [&](TypeDef iface)
+                        {
+                            default_iface_method_count = distance(iface.MethodList());
+                        });
+
+                        w.write(R"(
+protected unsafe % List<Func<IntPtr>> GetDefaultFactories()
```

Comment:
This is called multiple times by each parent class.  We should look at making it only constructed once.  We should also consider making this a readonly array or a readonly fixed capacity list as we know ahead of time the number of entries.